samtranslator/model/sam_resources.py

```diff
@@ -2300,9 +2302,55 @@ def _construct_ddb_datasources(
                 cfn_datasource.Description = cast(PassThrough, ddb_datasource.Description)
 
             cfn_datasource.DynamoDBConfig = self._parse_ddb_config(ddb_datasource)
-            datasources.append(cfn_datasource)
 
-        return datasources
+            cfn_datasource.ServiceRoleArn, role, connector_resources = self.parse_datasource_role(
+                ddb_datasource, cfn_datasource.get_runtime_attr("arn"), relative_id, kwargs
+            )
+
+            if role:
+                resources.append(role)
+
+            if connector_resources:
+                resources.extend(connector_resources)
+
+            resources.append(cfn_datasource)
+
+        return resources
+
+    def parse_datasource_role(
```

Comment:
Prefix with `_` so it's private by convention, otherwise consumers might be too eager to depend on it.

Reply:
Nice catch, that is definitely a typo thanks